### PR TITLE
cylc gui: enabled multiple selection in tree view

### LINKED ISSUE #1536: cylc gui: text/tree view: multiple line select and right-click menu
We should be able to select multiple lines in the tree view at once and launch a shared right click menu to e.g. trigger all tasks.

### PR DESCRIPTION
**Pull request by oliver-sanders**
Closes #1536 

For multiple selection the right click context menu yields all the normal items minus:
- Browse task URL
- View
- Trigger (edit run)

For multiple selection the following right click items are greyed out unless the action is valid for all tasks in the selection:
- Trigger (run now)
- Poll
- Kill

@hjoliver Please Review
@benfitzpatrick Please Review

### CODE REVIEW

**Comment by hjoliver** [comment]
Note this partially addresses #1871.

File: lib/cylc/gui/view_tree.py [hunk 1]
```diff
@@ -148,46 +148,49 @@ def on_treeview_button_pressed(self, treeview, event):
...
-                return False
-            task_state = self.t.state_summary[task_id]['state']
-            submit_num = self.t.state_summary[task_id]['submit_num']
+        path, col, _, _ = pth
+        tvte = TreeViewTaskExtractor(treeview)
+        selected_paths = [row[0] for row in tvte.get_selected_rows()]
+        if path not in selected_paths:
+            treeview.set_cursor(path, col, 0)
+
+        # Poulate lists of task info from the selected tasks.
```
Comment (hjoliver) [inline lib/cylc/gui/view_tree.py]: "Populate"

**Comment by hjoliver** [comment]
Review 1 - good, tests as working, no problems found.

File: lib/cylc/gui/app_gcylc.py [hunk 2]
```diff
@@ -1200,166 +1200,198 @@ def view_task_info(self, w, e, task_id, choice):
...
             gtk.STOCK_MEDIA_PLAY, gtk.ICON_SIZE_MENU)
         trigger_now_item.set_image(img)
-        items.append(trigger_now_item)
+        menu.append(trigger_now_item)
         trigger_now_item.connect(
-            'activate', self.trigger_task_now, task_id, task_is_family)
+            'activate', self.trigger_task_now, task_ids)
         trigger_now_item.set_sensitive(
-            t_state in TASK_STATUSES_TRIGGERABLE)
+            reduce(
```
Comment (benfitzpatrick) [inline lib/cylc/gui/app_gcylc.py]: Probably better to use `all`.

File: lib/cylc/gui/view_tree.py [hunk 3]
```diff
@@ -377,3 +380,90 @@ def click_exit(self, foo):
...
+        """Returns a list of rows that are currently selected in the provided
+        treeview. Rows are returned in the form (path, col1, col2)"""
+        model, rows = self.treeview.get_selection().get_selected_rows()
+        rows.sort()
+        _iter = model.get_iter_first()
+        _iter_ind = (0,)
+        ret = []
+        for row in rows:
+            # For each selected row.
+            while _iter_ind < row:
```
Comment (benfitzpatrick) [inline lib/cylc/gui/view_tree.py]: You can use `_iter = TreeModel.get_iter(row)`, but good effort!

**Comment by oliver-sanders** [comment]
@benfitzpatrick Addressed comments.

**Pull request merged**